### PR TITLE
add, update and uniform tooltips for Arch preferences

### PR DESCRIPTION
**Pull request by donovaly**
Add, update and uniform tooltips for Arch preferences.
Fixes issue [#3868](https://www.freecadweb.org/tracker/view.php?id=3868).

### CODE REVIEW

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 1]
```diff
@@ -185,7 +190,8 @@
         <item>
          <widget class="Gui::PrefLineEdit" name="lineEdit_2">
           <property name="toolTip">
-           <string>Only subtypes of this element will be imported. Keep value as &quot;IfcProduct&quot; to import all building elements.</string>
+           <string>Only subtypes of the specified element will be imported.
+Keep the element IfcProduct to import all building elements. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 2]
```diff
@@ -57,6 +58,10 @@
       </item>
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_6">
+        <property name="toolTip">
+         <string>Clones are used when objects have shared geometry
+One object is the base object, the others are clones. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 3]
```diff
@@ -308,7 +317,7 @@
         <item>
          <widget class="Gui::PrefLineEdit" name="gui::preflineedit">
           <property name="toolTip">
-           <string>A comma-separated list of Ifc entities to exclude from import</string>
+           <string>Comma-separated list of IFC entities to be excluded from imports </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 4]
```diff
@@ -533,7 +566,8 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_19">
         <property name="toolTip">
-         <string>When exporting an IFC file, if no site is found in the FreeCAD document, a default one will be added. A site is not mandatory by the IFC standard, but it is a common practice to always have at least one in the file.</string>
+         <string>If no site is found in the FreeCAD document, a default one will be added.
+A site is not mandatory but a common practice to have at least one in the file.</string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: A site is not mandatory but a common practice __is__ to have at least one in the file

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 5]
```diff
@@ -482,7 +508,8 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_15">
         <property name="toolTip">
-         <string>If this is checked, when possible, similar entities will be used only once in the file. This can reduce the file size a lot, but will make it less easily readable by humans</string>
+         <string>When possible, similar entities will be used only once in the file if possible.
+This can reduce the file size a lot, but will make it less easily readable. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 6]
```diff
@@ -466,7 +491,8 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_13">
         <property name="toolTip">
-         <string>If this is checked, all FreeCAD object properties will be stored inside the exported objects, allowing to recreate a full parametric model on reimport.</string>
+         <string>All FreeCAD object properties will be stored inside the exported objects,
+allowing to recreate a full parametric model on reimport. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 7]
```diff
@@ -434,7 +453,10 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_10">
         <property name="toolTip">
-         <string>Warning, experimental feature!</string>
+         <string>IFCOpenShell is a library that allows to import IFC files.
+Its serializer functionality allows to give it an OCC shape and it will
+produce adequate IFC geometry: NURBS, faceted, or anything else
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Period after the word `else`

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 8]
```diff
@@ -415,10 +432,12 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_9">
         <property name="toolTip">
-         <string>When exporting objects without UID, the generated UID will be stored inside the FreeCAD object for reuse next time that object is exported, which gives smaller diffs between versions</string>
+         <string>When exporting objects without unique ID (UID), the generated UID
+will be stored inside the FreeCAD object for reuse next time that object
+is exported. This leads to smaller differences between file versions. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 9]
```diff
@@ -399,7 +414,9 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_5">
         <property name="toolTip">
-         <string>Curved shapes that cannot be represented as curves in IFC are decomposed into flat facets. If this is checked, some additional calculations are done to join coplanar facets.</string>
+         <string>Curved shapes that cannot be represented as curves in IFC
+are decomposed into flat facets.
+If this is checked, additional calculation is done to join coplanar facets. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 10]
```diff
@@ -367,7 +381,8 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox">
         <property name="toolTip">
-         <string>Some IFC viewers don't like objects exported as extrusions. Use this to force all objects to be exported as BREP geometry.</string>
+         <string>Some IFC viewers don't like objects exported as extrusions.
+Use this to force all objects to be exported as BREP geometry. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui [hunk 11]
```diff
@@ -329,7 +338,8 @@
       <item>
        <widget class="Gui::PrefCheckBox" name="checkBox_14">
         <property name="toolTip">
-         <string>Fit view during import on the imported objects. This will slow down the import, but one can watch the import.</string>
+         <string>Fit view during import on the imported objects.
+This will slow down the import, but one can watch the import. </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-dae.ui [hunk 12]
```diff
@@ -238,7 +244,7 @@
         <item>
          <widget class="Gui::PrefCheckBox" name="checkBox">
           <property name="toolTip">
-           <string>Allow second order</string>
+           <string>Allow a second order mesh </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-dae.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-dae.ui [hunk 13]
```diff
@@ -37,6 +37,9 @@
         </item>
         <item>
          <widget class="Gui::PrefDoubleSpinBox" name="doubleSpinBox">
+          <property name="toolTip">
+           <string>All dimensions in the file will be scaled with this factor </string>
```
Comment (luzpaz) [inline src/Mod/Arch/Resources/ui/preferences-dae.ui]: Remove trailing whitespace

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui (hunk 4 above)
Comment (donovaly) [reply src/Mod/Arch/Resources/ui/preferences-ifc.ui]: I am not a native speaker, I thought I could omit a second 'is'

**Comment by donovaly** [comment]
All requested fixes are now in

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui (hunk 7 above)
Comment (luzpaz) [reply src/Mod/Arch/Resources/ui/preferences-ifc.ui]: This wasn't addressed in 363f985

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui (hunk 6 above)
Comment (luzpaz) [reply src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Note addressed in 363f985

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui (hunk 8 above)
Comment (luzpaz) [reply src/Mod/Arch/Resources/ui/preferences-ifc.ui]: Not addressed in 363f985

**Comment by donovaly** [comment]
Now really all fixes should be in.

File: src/Mod/Arch/Resources/ui/preferences-ifc.ui (hunk 6 above)
Comment (donovaly) [reply src/Mod/Arch/Resources/ui/preferences-ifc.ui]: I am still a newbie in FC. Why are these trailing whitespaces important?
I enabled modifications by developers so if you like you can immediately commit such fixes.

GitHub tells me now also
"New! Suggest specific code changes that the pull request author or assignees can immediately commit. You will be attributed in the commit."

So if you would use this I could quickly commit it.

**Comment by luzpaz** [comment]
@donovaly i don't have commit access for master. So I sent your branch a PR in https://github.com/donovaly/FreeCAD/pull/1/files

**Comment by donovaly** [comment]
> So I sent your branch a PR in https://github.com/donovaly/FreeCAD/pull/1/files

many thanks, it is in. So now the PR should really be ready to be committed.

**Comment by yorikvanhavre** [comment]
Merged, thanks both!

**Pull request merged**